metadata-io/src/main/java/com/linkedin/metadata/search/aggregator/AllEntitiesSearchAggregator.java

```diff
@@ -136,7 +136,9 @@ public SearchResult search(@Nonnull List<String> entities, @Nonnull String input
 
     postProcessTimer.stop();
     return new SearchResult().setEntities(new SearchEntityArray(rankedResult))
-        .setNumEntities(numEntities)
+    /* Pages will not show beyond the maximum search results allowed by elastic search. This fix will not allow
+    users to traverse to higher pages when searching without selecting any entity type */ 
+        .setNumEntities(numEntities > _entitySearchService.maxResultSize() ? _entitySearchService.maxResultSize() : numEntities)
```

Comment:
A bit confused about why this part is required. In some cases, we can have many more than 10k if we are searching across entities right?

Reply:
Yes, you are correct, when searching across entities we could have more than 10k datasets. However, because of elastic search capability, no data will be returned after page 1000 (assuming we query 10 datasets per page). I believe there is a report for this already #4575 which recommends using a rest endpoint that goes through all entities in mysql but does not account for latency.  Also, there has not yet been a sustainable solution with the `scroll_id `or` search_after` elastic search parameters to pull higher data.

Reply:
This is true in the case of searching for a single entity, but I'm wondering about the case that spans across entities? 

Imagine you have 9,999 datasets matching a query, and 9,999 dashboards as well. Then you'd rather see the total as 9,999 + 9,999 instead of 10,000. 

I agree that when you select down to a single entity, you may not want to see ever greater than 10,000 but I believe this is already happening today

Reply:
For example I am seeing this in my own deployment. This to me is more useful than seeing 10,000. In some cases, its not accurate, however (in the case that certain entity counts exceed 10k) 

<img width="298" alt="Screen Shot 2022-12-15 at 4 42 40 PM" src="https://user-images.githubusercontent.com/17549204/207996194-12f46d06-4483-49f1-aaed-8d606f7fffe7.png">

Reply:
Ping here :)

Reply:
right! I am dealing with the case where the entity counts exceed 10k across all entities. I see your point on the usability of seeing the true total across all entities. I wonder if you can navigate to those higher page numbers beyond 10k datasets in your deployment?